activesupport/lib/active_support/callbacks.rb

```diff
@@ -437,8 +436,6 @@ def self.build(filter, callback)
           case filter
           when Symbol
             new(nil, filter, [], nil)
-          when String
-            new(nil, :instance_exec, [:value], compile_lambda(filter))
```

Comment:
Could we put the error here? Would that avoid the duplicated checks below? They feel a bit odd where they are now, because they're checking & complaining about the type of a value those methods are barely using.

Reply:
I'll try but I think this method is executed at runtime not at definition time. But to be fair I didn't tried yet.

Reply:
Yeah, that is the case. That method is not called in definition time

Reply:
Ah, hmm. Do we care? It was important for the deprecation, to report it at the best possible time... but now it's "just" an error. I see the value in not letting it fall through to a confusing NoMethodError on String, but it currently seems a lot more protective than we normally are for wrong argument types.

Reply:
Addressed in the Callback initializer.